src/components/power-sensor/client.ts

```diff
@@ -76,29 +80,23 @@ export class PowerSensorClient implements PowerSensor {
   }
 
   async getReadings(extra = {}) {
-    const readings: Record<string, any> = {};
-    try {
-      [readings['voltage'], readings['isAc']] = await this.getVoltage(extra);
-    } catch (error) {
-      if (!(error as Error).message.includes('Unimplemented')) {
-        throw error;
-      }
-    }
-    try {
-      [readings['current'], readings['isAc']] = await this.getCurrent(extra);
-    } catch (error) {
-      if (!(error as Error).message.includes('Unimplemented')) {
-        throw error;
-      }
-    }
-    try {
-      readings['power'] = await this.getPower(extra);
-    } catch (error) {
-      if (!(error as Error).message.includes('Unimplemented')) {
-        throw error;
-      }
+    const { powersensorService } = this;
+    const request = new GetReadingsRequest();
+    request.setName(this.name);
+    request.setExtra(Struct.fromJavaScript(extra));
+
+    this.options.requestLogger?.(request);
+
+    const response = await promisify<GetReadingsRequest, GetReadingsResponse>(
+      powersensorService.getReadings.bind(powersensorService),
+      request
+    );
+
+    const result: Record<string, unknown> = {};
+    for (const [key, value] of response.getReadingsMap().entries()) {
+      result[key] = value.toJavaScript();
     }
-    return readings as PowerSensorReadings;
+    return result;
```

Comment:
Why are we using `Promise<Record<string, unknown>>` here instead of the `PowerSensorReadings` type?

Reply:
because I thought PowerSensorReadings was a type we were using before there was an API for powersensor.GetReadings

Reply:
We relied on those types in the RC, so now we see these errors:

```/__w/rdk/rdk/web/frontend/src/components/power-sensor/index.svelte:27:5
Error: Type 'unknown' is not assignable to type 'number | undefined'. (ts)
    const readings = await powerSensorClient.getReadings();
    voltageValue = readings.voltage;
    currentValue = readings.current;


/__w/rdk/rdk/web/frontend/src/components/power-sensor/index.svelte:28:5
Error: Type 'unknown' is not assignable to type 'number | undefined'. (ts)
    voltageValue = readings.voltage;
    currentValue = readings.current;
    powerValue = readings.power;


/__w/rdk/rdk/web/frontend/src/components/power-sensor/index.svelte:29:5
Error: Type 'unknown' is not assignable to type 'number | undefined'. (ts)
    currentValue = readings.current;
    powerValue = readings.power;
  } catch (error) {
  ```
  
With these changes we can no longer rely on the SDK to return us `current`, `voltage`, or `power` as a part of the readings. With this change we no longer know if those fields exist on the response object, and since they are type to `unknown` we have no idea if they are numbers, strings, or objects.

Reply:
oh sorry I had no idea, are you seeing this issue with movement sensor as well? or is it still working

Reply:
I haven't even tested anything out, the RC is failing it's build checks right now with an SDK version bump. These changes are not passing our standard checks.